build.gradle

```diff
@@ -23,6 +22,30 @@ plugins {
 	id 'java' // for jd gen
 }
 
+def toolchainLanguageVersion = JavaLanguageVersion.of(11)
+
+if (!JavaVersion.current().isCompatibleWith(JavaVersion.toVersion(toolchainLanguageVersion))) {
+	logger.warn ":The current Java Version ${JavaVersion.current()} is too old."
+	logger.warn ":Gradle may download newer Java online and the build may be slower."
```

Comment:
I think this is good for now, we can look at enforcing java 11 in a while, possibly along with engima.